platform/viewer/src/studylist/StudyListRoute.js

```diff
@@ -78,55 +131,54 @@ function StudyListRoute(props) {
     setActiveModalId('DicomStorePicker');
   }
 
-  // Called when relevant state/props are updated
-  // Watches filters and sort, debounced
-  useEffect(
-    () => {
-      const fetchStudies = async () => {
-        try {
-          setSearchStatus({ error: null, isSearchingForStudies: true });
-
-          const response = await getStudyList(
-            server,
-            debouncedFilters,
-            debouncedSort,
-            rowsPerPage,
-            pageNumber,
-            displaySize
-          );
-
-          setStudies(response);
-          setSearchStatus({ error: null, isSearchingForStudies: false });
-        } catch (error) {
-          console.warn(error);
-          setSearchStatus({ error: true, isFetching: false });
-        }
-      };
+  const fetchData = useCallback(
+    debounce(async newStudyListSessionProps => {
+      setSearchStatus({ error: null, isSearchingForStudies: true });
+      const {
+        sort,
+        filters,
+        rowsPerPage,
+        pageNumber,
+      } = newStudyListSessionProps;
+
+      const response = await getStudyList(
+        server,
+        filters,
+        sort,
+        rowsPerPage,
+        pageNumber,
+        displaySize
+      );
+
+      setStudies(response);
+      setSearchStatus({ error: null, isSearchingForStudies: false });
+    }, FETCH_DEBOUNCE_TIME),
+    [server, displaySize]
+  );
```

Comment:
We could leverage our `useDebounce` on a more complex `filters + sort` object. When the debounce elapses, we could fetchData and store/cache the filter + sort in session storage?

Reply:
yes we could by to preserve the current user experience flow its kept as its.
To do such change the effort will be higher.
Currently the flow is: child event (ex: key type on filter field), tell parent, do some stuff on parent, update child. I.e parent is totally responsible to process things and update child. With such implementation the updating of child is preserved, session always has the real data, but fetch is called only sometimes..

Totally makes sense to be the way you mentioned but, there are two problems I see:
- Integration between different child. I.e. pieces need to be in sync: When fetching list and page must be updated, ...
- It will cause some refactoring on child end.